### PR TITLE
Feature: Env check

### PR DESCRIPTION
**Pull request by nolostra**
[Feature: Env check](https://github.com/code100x/job-board/issues/353#issue-2529194745)

### CODE REVIEW

**Comment by aakash2330** [comment]
please distinguish b/w server and client envs  , the library let's you define them both seperately , and add empty example config for staging and production

File: src/env.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { createEnv } from '@t3-oss/env-nextjs';
+import { z } from 'zod';
+
+export const env = createEnv({
+  server: {
+    DATABASE_URL: z.string().min(1),
```
Comment (aakash2330) [inline src/env.ts:6]: custom zod validations for each env e.g. min length for secret might be different , POSTGRES_URL should be a url as such

File: package.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "name": "100xdevs-job-board",
   "version": "0.1.0",
   "private": true,
+  "type": "module",
```
[the remark this answers is not in the document]
Comment (nolostra) [reply package.json:5]: error was getting thrown for next config

**Comment by nolostra** [comment]
> please distinguish b/w server and client envs , the library let's you define them both seperately , and add empty example config for staging and production

ill make the necessary changes

File: src/env.ts (hunk 1 above)
Comment (nolostra) [reply src/env.ts:6]: cool

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (aakash2330) [reply package.json:5]: can you share the error

**Comment by aakash2330** [comment]
Screenshot 2024-09-20 at 3 44 02 PM

@nolostra some import error , can you check .

**Comment by nolostra** [comment]
> Screenshot 2024-09-20 at 3 44 02 PM
> @nolostra some import error , can you check .

looking into it

**Comment by nolostra** [comment]
> Screenshot 2024-09-20 at 3 44 02 PM
> @nolostra some import error , can you check .

sir node modules install karlo ek baar

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (nolostra) [reply package.json:5]: image

above image if i remove module sections


image

with including module

**Comment by VineeTagarwaL-code** [comment]
wohoo great discussions, great to see advancement thanks @aakash2330

**Pull request merged**